### PR TITLE
fix typos in javascript documentation

### PR DESCRIPTION
**Pull request by garanews**
fix typos in javascript documentation

### CODE REVIEW

File: content/javascript/concepts/number-methods/number-methods.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 Title: 'Number Methods'
-Description: 'JavaScript treats primitive values as objects when its executing methods and properties. The Number object is used to represent numbers, like 3.14, -10, or 2022, and it comes with a handful of methods thats useful for handling with numbers.'
+Description: 'JavaScript treats primitive values as objects when its executing methods and properties. The Number object is used to represent numbers, like 3.14, -10, or 2022, and it comes with a handful of methods that is useful for handling with numbers.'
```
Comment (SSwiniarski) [inline content/javascript/concepts/number-methods/number-methods.md:3]:
```suggestion
Description: 'JavaScript treats primitive values as objects when its executing methods and properties. The Number object is used to represent numbers, like 3.14, -10, or 2022, and it comes with a handful of methods that are useful for handling with numbers.'
```

File: content/javascript/concepts/sort/sort.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 Title: 'Sort'
-Description: 'In JavaScript, the .sort() method of an array sorts the contents of an array and returns the sorted array. This sorting is done in place and affects the original array. No copy is made. The default sort is in ascending string order. The .sort() method allows the passing of a comparison function to change the ordering of the sort. javascript // Preform the default sort somearray.sort() // Perform the sort using an arrow function for comparisons somearray.sort((A, B) => { ... } )'
+Description: 'In JavaScript, the .sort() method of an array sorts the contents of an array and returns the sorted array. This sorting is done in place and affects the original array. No copy is made. The default sort is in ascending string order. The .sort() method allows the passing of a comparison function to change the ordering of the sort. javascript // Perform the default sort somearray.sort() // Perform the sort using an arrow function for comparisons somearray.sort((A, B) => { ... } )'
```
Comment (SSwiniarski) [inline content/javascript/concepts/sort/sort.md:3]:
```suggestion
Description: 'Sorts the contents of an array and returns the sorted array. This sorting is done in place and affects the original array.'
```

File: content/javascript/concepts/arrays/terms/join/join.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ Elements of an array are converted to strings and concatenated together, returni
 ## Syntax
 
 ```js
-array.join(seperator);
+array.join(separator);
 ```
 
 An optional separator parameter specifies a string to separate each pair of adjacent element of the array.
```
Comment (Dusch4593) [inline content/javascript/concepts/arrays/terms/join/join.md:23]:
```suggestion
An optional `separator` parameter specifies a string or character that goes between each array element in the returned string.
```

File: content/javascript/concepts/number-methods/terms/toExponential/toExponential.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ CatalogContent:
   - 'paths/front-end-engineer-career-path'
 ---
 
-The `.toExponential()` method converts a number to a string of its exponetial form.
+The `.toExponential()` method converts a number to a string of its exponential form.
```
Comment (Dusch4593) [inline content/javascript/concepts/number-methods/terms/toExponential/toExponential.md:15]:
```suggestion
The **`.toExponential()`** method converts a number to a [string](https://www.codecademy.com/resources/docs/javascript/strings) of its exponential form.
```

File: content/javascript/concepts/number-methods/terms/toPrecision/toPrecision.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ CatalogContent:
   - 'paths/front-end-engineer-career-path'
 ---
 
-The `.toPrecision()` method converts a number to a string of its specificied precision form.
+The `.toPrecision()` method converts a number to a string of its specified precision form.
```
Comment (Dusch4593) [inline content/javascript/concepts/number-methods/terms/toPrecision/toPrecision.md:15]:
```suggestion
The **`.toPrecision()`** method converts a number to a [string](https://www.codecademy.com/resources/docs/javascript/strings) of its specified precision form.
```

File: content/javascript/concepts/number-methods/terms/toPrecision/toPrecision.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ---
 Title: '.toPrecision()'
-Description: 'Convert a number to a string of its specificied precision form.'
+Description: 'Convert a number to a string of its specified precision form.'
```
Comment (Dusch4593) [inline content/javascript/concepts/number-methods/terms/toPrecision/toPrecision.md:3]:
```suggestion
Description: 'Converts a number to a string of its specified precision form.'
```

File: content/javascript/concepts/promise/terms/any/any.md [hunk 7]
```diff
@@ -23,7 +23,7 @@ The `iterableObject` is usually an array of Promise objects. If the array is emp
 
 ## Example #1
```
Comment (Dusch4593) [inline content/javascript/concepts/promise/terms/any/any.md:24]:
```suggestion
## Example 1
```

Comment (Dusch4593) [reply content/javascript/concepts/promise/terms/any/any.md:24]: Let's make the same edit for the other header on line # 35.

File: content/javascript/concepts/sort/sort.md [hunk 8]
```diff
@@ -20,7 +20,7 @@ The `.sort()` method allows the passing of a comparison function to change the o
 ## Syntax
 
 ```javascript
```
Comment (Dusch4593) [inline content/javascript/concepts/sort/sort.md:22]:
```suggestion
```pseudo
```

File: content/javascript/concepts/sort/sort.md [hunk 9]
```diff
@@ -20,7 +20,7 @@ The `.sort()` method allows the passing of a comparison function to change the o
...
-// Preform the default sort
+// Perform the default sort
 somearray.sort()
```
Comment (Dusch4593) [inline content/javascript/concepts/sort/sort.md:24]:
```suggestion
someArray.sort()
```

Let's make the same edit for the other lines in this snippet.

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@garanews Finished with my review and left some comments/suggestions. I'd also recommend the following:

For all entries in this PR:
- Let's make sure the first reference of a concept or term is emphasized in **bold**.
- Let's ensure all code blocks under the "Syntax" section is a \```pseudo snippet.

Under `ajax.md`:
- Link "server" to the following Docs entry: https://www.codecademy.com/resources/docs/general/server
- Let's remove the canonical `<link/>` on line # 43 since it is already referenced on line # 17.

Under `join.md`:
- On line # 15, let's rephrase to the following:
```
The **`.join()`** method returns a [string](https://www.codecademy.com/resources/docs/javascript/strings) representation of array elements concatenated together.
```
- On lines # 29 and 35, let's remove the underscores from "separator."

Under `arrow-functions.md`:
- Add `'ES6'` to the list of `Tags` (lines # 7 - 10).
- On line # 16, replace the relative path for JS functions with the absolute path: `https://www.codecademy.com/resources/docs/javascript/functions`.
- Start a new line on # 36 and insert a new lv. 2 header called "## Example".
- On line # 69, let's link "`this`" to the JS entry for it: https://www.codecademy.com/resources/docs/javascript/this

Under `size.md`:
- Let's add `'Properties'` to the list of `Tags` (lines # 7-9).
- On line # 15, let's rephrase to the following:
```
The **.size** property returns the number of entries in a `Map` object.
```

Under `number-methods.md`:
- Let's add `'Objects'` to the list of `Tags` (lines # 7-10).
- I would rearrange the content in lines # 16 - 18 such that the concept "`Number` object" is mentioned in the entry, first (and **bolded**, of course).

Under `toExponential.md`:
- On line # 47, let's change Codebyte signature to \```codebyte/javascript.

Under `toPrecision.md`:
- On line # 50, let's change Codebyte signature to \```codebyte/javascript.

Under `allSettled.md`:
- I would change "Example 1" to just "Example".
- Then I would change "Example 2" to "Codebyte Example" and convert the code block to a \```codebyte/javascript snippet. (just tested the code and it runs! 😄)
- Lastly, because the Codebyte is viable, I would suggest removing all content from line # 82 to the end of the file.

Under `any.md`:
- On line # 14, let's rephrase to the following:
```
The **`.any()`** method iterates over an `iterableObject` of Promises ...
```

Under `regexp.md`:
- Let's change the code block under "Syntax" to a \```pseudo snippet.
- Let's change the remaining code blocks from \```javascript to just \```js.

Under `sort.md`:
- Let's change the non-syntax code blocks from \```javascript to just \```js.

Under `this.md`:
- Let's change the remaining code blocks from \```javascript to just \```js.
- On line # 186, let's link "arrow function" to the Docs entry for it: https://www.codecademy.com/resources/docs/javascript/arrow-functions

File: content/javascript/concepts/arrow-functions/arrow-functions.md [hunk 10]
```diff
@@ -1,6 +1,6 @@
...
+Description: 'Arrow function expressions were introduced in ES6. These expressions are a clean and concise alternative to the traditional function syntax. The syntax for an arrow function expression does not require the function keyword and uses a fat arrow => to separate the parameter(s) from the body. However, they are limited and cannot be used in all situations. There are several variations of arrow functions: - Arrow functions with a single parameter do not require () around the parameter list. - Arrow functions with a single expression can use the concise function body which returns the result of the expression without the return keyword. Arrow function with no arguments: js const printHello = () => {'
```
[the remark this answers is not in the document]
Comment (garanews) [reply content/javascript/concepts/arrow-functions/arrow-functions.md:3]: for me no problem to cut description, so if you say ok, I cut it.

File: content/javascript/concepts/npm/npm.md [hunk 11]
```diff
@@ -1,6 +1,6 @@
...
+Description: 'Node Package Manager (NPM) is a large collection of JavaScript packages. This makes adding complexity and funcionality to JavaScript project seamless. NPM is also the official package manager of Node.js. Therefore, if Node.js was previously downloaded, NPM is already installed! Packages are tools composed of files and folders of JavaScript code. They provide one or more operations that can be installed, imported and used in a separate project. Thanks to NPM, the installation process for packages is uniform. NPM comes as two primary tools: - A command-line tool with abilities that include installing and uninstalling packages. - An online registry of all packages registered with NPM. On the NPM registry, all public packages can be searched and reviewed before installing. Most packages come with a README.md file with instructions for installation and usage. As a command-line tool, NPM comes with a litany of commands for working with packages. One of the most common NPM commands is:'
```
[the remark this answers is not in the document]
Comment (garanews) [reply content/javascript/concepts/npm/npm.md:3]: for me no problem to cut description, so if you say ok, I cut it.

File: content/javascript/concepts/sort/sort.md (hunk 2 above)
Comment (garanews) [reply content/javascript/concepts/sort/sort.md:3]: for me no problem to cut description, so if you say ok, I cut it.

**Comment by garanews** [comment]
@Dusch4593 applied your suggested fixes, just waiting for @SSwiniarski acknowledge to cut the descriptions.

File: content/javascript/concepts/arrow-functions/arrow-functions.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (SSwiniarski) [reply content/javascript/concepts/arrow-functions/arrow-functions.md:3]: @garanews please do.

File: content/javascript/concepts/npm/npm.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (SSwiniarski) [reply content/javascript/concepts/npm/npm.md:3]: @garanews please do.

File: content/javascript/concepts/sort/sort.md (hunk 2 above)
Comment (SSwiniarski) [reply content/javascript/concepts/sort/sort.md:3]: @garanews please do.

**Review by SSwiniarski — APPROVED** [review]
Looks good to me.

**Review by Dusch4593 — APPROVED** [review]
@garanews LGTM, as well 👍🏻

I made those last needed edits (plus some other ones) in 40926daceee881a5a86975a8845f3fe102023867 but when this is merged, you will receive sole credit 😎

**Pull request merged**